src/script/page/RightSidebar/AddParticipants/AddParticipants.tsx

```diff
@@ -0,0 +1,334 @@
+
+  const [searchInput, setSearchInput] = useState<string>('');
+  const [contacts, setContacts] = useState<User[]>([]);
+  const [selectedContacts, setSelectedContacts] = useState<User[]>([]);
+
+  const [isInitialServiceSearch, setIsInitialServiceSearch] = useState<boolean>(true);
+
+  const enabledAddAction = selectedContacts.length > 0;
+
+  const headerText = selectedContacts.length
+    ? t('addParticipantsHeaderWithCounter', selectedContacts.length)
+    : t('addParticipantsHeader');
+
+  const showIntegrations = useMemo(() => {
+    const isServicesEnabled = isServicesRoom || isGuestAndServicesRoom;
+    const isService = !!firstUserEntity?.isService;
+    const allowIntegrations = isGroup || isService;
+
+    return isTeam && allowIntegrations && inTeam && !isTeamOnly && isServicesEnabled;
+  }, [isServicesRoom, isGuestAndServicesRoom, firstUserEntity, isGroup, isTeam, inTeam, isTeamOnly]);
+
+  const manageServicesUrl = getManageServicesUrl('client_landing');
+  const isSearching = searchInput.length > 0;
+
+  const onAddPeople = () => setCurrentState(PARTICIPANTS_STATE.ADD_PEOPLE);
+
+  const searchServices = async (value: string) => {
+    await integrationRepository.searchForServices(value);
+    setIsInitialServiceSearch(false);
+  };
+
+  const onAddServices = async () => {
+    setCurrentState(PARTICIPANTS_STATE.ADD_SERVICE);
+
+    await searchServices(searchInput);
+  };
+
+  const openManageServices = () => {
+    if (manageServicesUrl) {
+      safeWindowOpen(manageServicesUrl);
+    }
+  };
+
+  const onServiceSelect = (serviceEntity: ServiceEntity) => {
+    togglePanel(PanelViewModel.STATE.GROUP_PARTICIPANT_SERVICE, {
+      addMode: true,
+      entity: serviceEntity,
+    });
+  };
+
+  const addUsers = async () => {
+    const userEntities = selectedContacts.slice();
+
+    await conversationRepository.addUsers(activeConversation, userEntities);
+  };
+
+  const onAddParticipants = async () => {
+    await addUsers();
+    onBack();
+  };
+
+  const onSearchInput = async (value: string) => {
+    await searchServices(value);
+    setSearchInput(value);
+  };
+
+  const getUserEntities = useCallback(() => {
+    let userEntities: User[] = [];
+
+    if (isTeam) {
+      userEntities = isTeamOnly || isServicesRoom ? teamMembers.sort(sortUsersByPriority) : teamUsers;
+    } else {
+      userEntities = connectedUsers;
+    }
+
+    return userEntities.filter(userEntity => {
+      return participatingUserIds.find(userId => matchQualifiedIds(userEntity, userId));
+    });
+  }, [isTeam, isTeamOnly, isServicesRoom, teamUsers, connectedUsers, participatingUserIds]);
+
+  useEffect(() => {
+    const userEntities = getUserEntities();
+    setContacts(userEntities);
+  }, []);
+
+  return (
+    <div id="add-participants" className="add-participants panel__page panel__page--visible">
+      <PanelHeader
+        showBackArrow
+        goBackUie="go-back-add-participants"
+        onGoBack={onBack}
```

Comment:
Please talk to Astrid and confirm if you need to add these props for PanelHeader goBackTitle, title, closeBtnTitle